sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -594,7 +596,9 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
     // returns null if the product type doesn't have a primary constructor, e.g. HiveFunctionWrapper
     case p: Product => try {
```

Comment:
Thank you very much for your reply. 
Shall we also allow the JDBCRelation to be converted to JSON like the following?
```
     val jdbcDF = sqlContext.read.format("jdbc").options(
       Map("url" -> "jdbc:h2:mem:testdb0;user=testUser;password=testPass",
         "dbtable" -> "people")).load().queryExecution.logical.toJSON
```